docs/building-blocks/extending-templates/objects-as-parameters.md

```diff
@@ -0,0 +1,322 @@
+      "apiVersion": "2015-06-15",
+      "type": "Microsoft.Network/virtualNetworks",
+      "name": "[parameters('VNetSettings').name]",
+      "location":"[resourceGroup().location]",
+      "properties": {
+          "addressSpace":{
+              "addressPrefixes": [
+                  "[parameters('VNetSettings').addressPrefixes[0].addressPrefix]"
+              ]
+          },
+          "subnets":[
+              {
+                  "name":"[parameters('VNetSettings').subnets[0].name]",
+                  "properties": {
+                      "addressPrefix": "[parameters('VNetSettings').subnets[0].addressPrefix]"
+                  }
+              },
+              {
+                  "name":"[parameters('VNetSettings').subnets[1].name]",
+                  "properties": {
+                      "addressPrefix": "[parameters('VNetSettings').subnets[1].addressPrefix]"
+                  }
+              }
+          ]
+        }
+    }
+  ],          
+  "outputs": {}
+}
+
+```
+
+The corresponding parameter file looks like:
+
+```json
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters":{ 
+      "VNetSettings":{
+          "value":{
+              "name":"VNet1",
+              "addressPrefixes": [
+                  { 
+                      "name": "firstPrefix",
+                      "addressPrefix": "10.0.0.0/22"
+                  }
+              ],
+              "subnets":[
+                  {
+                      "name": "firstSubnet",
+                      "addressPrefix": "10.0.0.0/24"
+                  },
+                  {
+                      "name":"secondSubnet",
+                      "addressPrefix":"10.0.1.0/24"
+                  }
+              ]
+          }
+      }
+  }
+}
+```
+
+In this example all the values specified for the VNet come from a single parameter, `VNetSettings`. This is useful for property value management because you to keep all the values for a particular resource in single object. And while this example uses an object as a parameter, you can also use an array of objects as a parameter. You refer to the objects using an index into the array.
+
+## Use object instead of multiple arrays
+
+You may be familiar with using a copy loop to create multiple resources. Typically you provide an JSON array of resource properties and the copy loop iterates through it and selects values. This works well when creating multiple resources of the same type, but it can be troublesome when used to create child resources. 
+
+There's two reasons for this issue. First, Resource Manager attempts to deploy child resources in parallel, but your deployment fails when two child resources update the parent simultaneously. 
+
+Second, each property value array is iterated in parallel and if the shape of each array is not the same shape an error occurs. For example, consider the following property arrays:
```

Comment:
I'm confused by this part - Looks like the error is that firstProperty and secondProperty are arrays with different lengths, not that the shape (of the objects in the arrays?) is different. 

I think this pattern is saying that it's safer to loop over a single array of objects, rather than loop over several arrays in parallel? It doesn't seem related to the problem statement from the intro paragraph (limit of 255 parameters)